### PR TITLE
Add functional to check correct date

### PR DESCRIPTION
**Pull request by MiBBy-4**

### CODE REVIEW

File: app/controllers/api/v1/ideas_controller.rb [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
 
   def create
     customer_id = session[:customer_id]
     @idea = Idea.new(idea_params)
+    @idea.publication_period = today + 30
 
     if @idea.save
       render json: {
-        @idea, status: 201,
+        errors: @idea, status: 201,
```
Comment (AndreyAnischik) [inline app/controllers/api/v1/ideas_controller.rb:27]: no need to add errors if activity was saved

File: app/controllers/api/v1/ideas_controller.rb [hunk 2]
```diff
@@ -42,8 +50,26 @@ def destroy
     @idea.destroy
   end
 
+  def update_publication_period
+    @idea.publication_period += 10
+     if @idea.save
+      render json: {
+        status: 200
```
Comment (AndreyAnischik) [inline app/controllers/api/v1/ideas_controller.rb:57]: idea

File: app/controllers/api/v1/ideas_controller.rb [hunk 3]
```diff
@@ -42,8 +50,26 @@ def destroy
     @idea.destroy
   end
 
+  def update_publication_period
+    @idea.publication_period += 10
```
Comment (AndreyAnischik) [inline app/controllers/api/v1/ideas_controller.rb:54]: magic number

File: config/routes.rb [hunk 4]
```diff
@@ -11,6 +11,7 @@
   namespace :api do
     namespace :v1 do
       resources :ideas
+      patch 'publication_period_update/:id', to: 'ideas#update_publication_period'
```
Comment (AndreyAnischik) [inline config/routes.rb:14]: use member helper

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/ideas_controller.rb (hunk 1 above)
Comment (MiBBy-4) [reply app/controllers/api/v1/ideas_controller.rb:27]: Yep, sorry. My bad.

File: app/controllers/api/v1/ideas_controller.rb [hunk 5]
```diff
@@ -1,27 +1,38 @@
...
+        status: 403,
+        message: "Sorry, but Idea is unavailable"
+      }
+    end
   end
 
   def create
     customer_id = session[:customer_id]
     @idea = Idea.new(idea_params)
+    @idea.publication_period = today + 30
```
Comment (AndreyAnischik) [inline app/controllers/api/v1/ideas_controller.rb:25]: magic number

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

**Review by AndreyAnischik — APPROVED** [review]

**Pull request merged**